app/components/ui/molecules/SearchBox.test.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import { ThemeProvider } from 'styled-components'
+import { escapeRegExp } from 'lodash'
+import theme from '@elifesciences/elife-theme'
+
+import SearchBox from './SearchBox'
+
+const searchOptions = [
+  { value: 'first option' },
+  { value: 'final option' },
+  { value: 'second option' },
+]
+
+function filterFunction(options, searchValue, field) {
+  if (!searchValue) return options
+
+  const inputValue = searchValue.trim().toLowerCase()
+  if (!inputValue) return options
+
+  return options.filter(option =>
+    option[field].toLowerCase().includes(inputValue),
+  )
+}
+
+function getMatchIndex(inputValue, option) {
+  const re = new RegExp(escapeRegExp(inputValue))
+  const match = re.exec(option.toLowerCase())
+  if (match) return match.index
+  return -1
+}
+
+function makeWrapper() {
+  return mount(
+    <ThemeProvider theme={theme}>
+      <SearchBox
+        filterFunction={filterFunction}
+        getMatchIndex={getMatchIndex}
+        onSubmit={jest.fn()}
+        options={searchOptions}
+      />
+    </ThemeProvider>,
+  )
+}
+
+const wrapper = makeWrapper()
+
+describe('SearchBox component tests', () => {
+  it('renders an input field', () => {
+    expect(wrapper.find('input')).toHaveLength(1)
+  })
+
+  it('has search icon', () => {
+    expect(wrapper.find('#search-icon')).toHaveLength(1)
+  })
+
+  it("doesn't show any suggestions on empty input", () => {
+    wrapper.find('input').simulate('change', { target: { value: '' } })
+    expect(wrapper.find(SearchBox).instance().state.suggestions).toEqual([])
+  })
+
+  function compareSuggestions(searchValue) {
+    const expectedSuggestions = filterFunction(
+      searchOptions,
+      searchValue,
+      'value',
+    )
+    wrapper.find('input').simulate('change', { target: { value: searchValue } })
+    const receivedSuggestions = wrapper.find(SearchBox).instance().state
+      .suggestions
+    expect(receivedSuggestions).toEqual(expectedSuggestions)
+  }
+
+  it('typing something will show suggestions based on filtering function, test 1', () => {
+    compareSuggestions('option')
+  })
+
+  it('typing something will show suggestions based on filtering function, test 2', () => {
+    compareSuggestions('f')
+  })
+
+  it('typing something will show suggestions based on filtering function, test 3', () => {
+    compareSuggestions('xyz')
```

Comment:
Shouldn't this fail?

If our options are:
```
  { value: 'first option' },
  { value: 'final option' },
  { value: 'second option' },
```

then shouldn't typing in "xyz" actually show no suggestions at all? :confused:

Reply:
Ah no, I see, it's the title of the test that's misleading.

It says "typing something will show suggestions based on filtering function", but the assertion is actually checking whether the received suggestions are the same as the expected suggestions. Since we expect no results & we receive no results, the tests passes.

So perhaps the issue is just the naming of the test?

Reply:
I think the title is true, but maybe the wording is not the best. However, I wouldn't consider this of high importance, if a test fails you usually jump to the test after its description and see what's up.

Reply:
Hmm I'm not so sure. The title "typing something will show suggestions" would imply to me that the test covers whether something is being rendered. But as you said above, we haven't quite got this working with enzyme. So, as it is, the suggestions are _not_ shown. In fact, this is only testing whether the state is updated & I reckon the name should reflect that?

I would say that we need to figure out how to make this work with enzyme, otherwise we're not able to actually test that the dropdown functions as it's supposed to. But I struggled with this exact thing in enzyme myself, when writing the SubjectAreaDropdown tests, so I totally get why you've opted to test the state change alone & not whether it renders.

Reply:
Updating the name doesn't change what you're testing. It just more accurately describes what this test currently does.

If you think that testing whether the state has been updated is a bad test, then the test needs to be changed, no? Because that's exactly what it's doing, isn't it?

Reply:
Not if you make a reasonable assumption and plan on changing it for the better in the future. I've seen plenty of similar assumptions in tests before, I'm just the only one drawing attentions towards it so others are aware.